### ISSUE TITLE
cd ffwme >> gulp Causes "TypeError: Cannot read property 'prototype' of undefined

### ISSUE THREAD

**Issue by cScarlson** [ffwdme/ffwdme.js#12]
After pulling down ffwme.js, running the `gulp` command in the root causes a type error:

```
util.js:634
    ctor.prototype = Object.create(superCtor.prototype, {
at ffwme\node_modules\browser-sync\node_modules\http-proxy\lib\http-proxy\index.js:111:17
...
at Module._compile (module.js:460:26)
```

Please see this Stack Overflow article, as this pertains to the [eventemitter3] `EventEmitter` (`EE3`) library:
http://stackoverflow.com/questions/31115494/error-starting-express-proxy-server

Tried cd'ing into `/gulp/tasks/browserify` and running `gulp` -- same error. Cannot add ffwme /core files in directly either as this causes a `TypeError` from invoking `require`.

Wasn't able to get started :(

Either way, it sounds like to 'get started' we need to run `gulp` -- is it actually necessary, or is there a `/build/core.js` that I'm just missing? Otherwise, we will have to write our own tasks (gulp or not) to employ _browserify_ to avoid the aforementioned `require(...)` invocation errors.

Also, in the "Getting Started" docs, would be nice to see which files should be pulled in; based upon user needs (basic | advanced | etc).

Thanks -- would _love_ to hear quick feedback on this progress as I'm on a deadline for a Proof-Of-Concept.

Cheers,

Cody

**Comment by fabrik42** [comment]
Ok - I think you have some valid points, but I need to take some time to look into them.

If you just want to "use" the lib without building it yourself, feel free to use the compiled versions here:

http://dev.ffwdmejs.org/ffwdme-core.js
http://dev.ffwdmejs.org/ffwdme-components.js
http://dev.ffwdmejs.org/ffwdme-debug.js

**Comment by cScarlson** [comment]
fabrik24, thank you so much -- thank you for taking the time to look into this; the provided links will get me where I need to be for a while.

Much Appreciation,

Cody

**Comment by fabrik42** [comment]
Hi Cody,

I just updated some dependencies and it would be cool, if you could try again to get your development server running.
- Please clone the newest version of the repo or update your existing one
- `cd` into the main directory of this repository on your machine
- Delete your `node_modules` folder.
- Call `npm install` - this should succeed without errors
- Call `gulp` - this should open a browser window on `localhost:3000` and the file watcher should run in your terminal window.

Cheers,

Christian

**Comment by fabrik42** [comment]
Please note, that I currently use node v5.0.0

**Comment by cScarlson** [comment]
Hey Christian,

I really appreciate you tackling this -- in such a timely manner, too. I'm definitely running an older version of Node, but I'll be sure to dive into it tomorrow as to keep as tight a feedback loop for you as possible.

Thanks again,

Cody

**Comment by cScarlson** [comment]
Just a quick update,

As I followed the directions provided above, I _seem_ to have had more success this time around; but still was not able to lift off. When I cd'd into `/ffwme` and ran `npm install` I had a lot of errors in my console -- most-to-all looking for Python on my machine (which I don't currently have).

Either way, I took a poke in the dark and tried running `gulp`. I received an error as it was looking for a `secrets.json` file, which I created in the root, and added the example content from `example_secrets.json` --> `gulp` and got `undefined is not a function` -- whose traceback looks something like this:

```
\gulp\tasks\browserify.js:46
[others circumvolving \orchastrator\*\runTask.js|index.js]
```

I hope the above makes sense and provides some insight.

However, I think I'm on a much older version of Node, so when I have some more time, I'm going to update Node and try again.

If its any constellation, I've moved out of the timebox I was in for my previously mentioned POC/Prototype; as so, my needs are less urgent then they were before -- so I hope this lessens the pressure you guys put on yourselves to move through this issue.

Thank you Christian -- and any others -- who took so much attention to this issue, and I'll be tracking its progress over the next 3 weeks to jump back into things.

Cheers,

Cody
